src/System.Private.CoreLib/shared/System/TimeSpan.cs

```diff
@@ -239,11 +239,10 @@ private static TimeSpan Interval(double value, int scale)
         {
             if (double.IsNaN(value))
                 throw new ArgumentException(SR.Arg_CannotBeNaN);
-            double tmp = value * scale;
-            double millis = tmp + (value >= 0 ? 0.5 : -0.5);
+            double millis = value * scale;
```

Comment:
Instead of multiplying first by `scale` and then again by `TicksPerMillisecond`, you can do just one multiplication by folding `TicksPerMillisecond` into scale.